### ISSUE TITLE
Button component enhancements to support `<a>` elements

### ISSUE THREAD

**Issue by jgiardino** [patternfly/patternfly-next#402]
The Button component example should be updated to include the following:

- Examples that show `<a>` elements being used instead of `<button>` elements. 
- Updated Accessibility documentation to include attributes that would be used for `<a>` instead of `<button>`. For reference, the following attributes are documented for the Dropdown component:

Attribute | Applied | Outcome
-- | -- | --
disabled="disabled" | button.pf-c-dropdown__menu-item | When the menu item uses a button element, indicates that it is unavailable and removes it from keyboard focus
aria-disabled="true" | a.pf-c-dropdown__menu-item | When the menu item uses a link element, indicates that it is unavailable
tabindex="-1" | a.pf-c-dropdown__menu-item | When the menu item uses a link element, removes it from keyboard focus
